File: in_Java/1302 Deepest Leaves Sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        if(root == null) return 0;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        TreeNode cur;
        int n;
        List<Integer> L = new ArrayList<>();
        while(q.size() != 0){
            n = q.size();
            L = new ArrayList<>();
            while(n-- != 0){
                cur = q.poll();
                L.add(cur.val);
                if(cur.left != null) q.add(cur.left);
                if(cur.right != null) q.add(cur.right);
            }
        }
        int res = 0;
        for(int i : L) res += i;
        return res;
    }
}

File: in_Python/1302 Deepest Leaves Sum.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def deepestLeavesSum(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        res = []
        q = []
        if root == None:
            return 0
        q.append(root)
        while len(q) != 0:
            n = len(q)
            rowsum = 0
            for i in range(n):
                h = q.pop(0)
                if h.left != None:
                    q.append(h.left)
                if h.right != None:
                    q.append(h.right)
                rowsum += h.val
            res.append(rowsum)
        return res[-1]